test/run_pass/test_net_http_request_response.js

```diff
@@ -0,0 +1,206 @@
+/* Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var assert = require('assert');
+var http = require('http');
+var net = require('net');
+
+// Messages for further requests.
+var message = 'Hello IoT.js';
+
+// Options for further requests.
+var options = {
+  method: 'POST',
+  port: 3005,
+  path: '/',
+  headers : {'Content-Length': message.length}
+};
+
+
+var server0 = http.createServer(function(request, response) {
+  response.end('default');
+});
+server0.listen(80, 5);
+
+// Simple get request without options parameter.
+// Every options should be the default:
+// the default port is 80 and the default method is GET.
+var request0 = http.get(function(response) {
+  var str = '';
+
+  response.on('data', function(chunk) {
+    str += chunk;
+  });
+
+  response.on('end', function() {
+    assert.equal(str, 'default');
```

Comment:
In the case, it seems right to check if the api doesn't crash, or else if it throws any information for the wrong usage. However, assert here checks str although it's even never be called. Confusing.